src/main/java/org/onebusaway/alexa/lib/ObaUserClient.java

```diff
@@ -35,18 +37,39 @@
 public class ObaUserClient {
     private static final int DEFAULT_SEARCH_RADIUS_METERS = 40000;
 
-    public ObaUserClient(String endpoint) {
-        log.debug("Instantiating ObaUserClient with endpoint " + endpoint);
+    public ObaUserClient(String obaBaseUrl) {
+        log.debug("Instantiating ObaUserClient with obaBaseUrl " + obaBaseUrl);
         try {
-            ObaApi.getDefaultContext().setBaseUrl(
-                    String.format("http://api.%s.onebusaway.org/", endpoint));
+            ObaApi.getDefaultContext().setBaseUrl(obaBaseUrl);
         } catch (URISyntaxException e) {
-            log.error("OBA Endpoint we constructed was invalid: " + e.getMessage());
+            log.error("ObaBaseUrl we constructed was invalid: " + e.getMessage());
         } catch (Exception e) {
-            log.error("Exception setting OBA base URL: " + e.getMessage());
+            log.error("Exception setting ObaBaseUrl: " + e.getMessage());
             throw e;
         }
     }
+
+    /**
+     * Get the closest region to a given location.  Regions come from the Regions API
+     * (http://regions.onebusaway.org/regions-v3.json), which is a centralized directory of all known
+     * OneBusAway servers.  Each region has it's own OBA REST API endpoint
+     * (http://developer.onebusaway.org/modules/onebusaway-application-modules/current/api/where/index.html),
+     * which we will then use to get stop and arrival data for that region.
+     *
+     * @param l geographic location used to search for nearby regions
+     * @return the closest region in the Regions API to the given location, or null if there are no nearby regions
+     * (within 100 miles of the provided location) or a region couldn't be found.
+     */
+    public static Optional<ObaRegion> getClosestRegion(Location l) {
+        log.debug("Invoked getClosestRegion() with location " + l.toString());
+        ObaRegionsResponse response = ObaRegionsRequest.newRequest().call();
+        ArrayList<ObaRegion> regions = new ArrayList<>(Arrays.asList(response.getRegions()));
+        return Optional.of(RegionUtils.getClosestRegion(
```

Comment:
Since RegionUtils.getClosestRegion can return null, you need `Optional.ofNullable` instead of `Optional.of`. `of` will raise an exception if you give it null.